packages/in_app_purchase/test/store_kit_wrappers/sk_product_request_wrapper_test.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:test/test.dart';
+import 'package:in_app_purchase/src/store_kit_wrappers/sk_product_request_wrapper.dart';
+import 'package:in_app_purchase/src/channel.dart';
+import 'package:flutter/services.dart';
+import 'package:in_app_purchase/src/in_app_purchase_connection/product.dart';
+import '../fake_platform_views_controller.dart';
+
+void main() {
+  final FakePlatformViewsController fakePlatformViewsController =
+      FakePlatformViewsController();
+
+  final Map<String, dynamic> subMap = <String, dynamic>{
+    'numberOfUnits': 0,
+    'unit': 1
+  };
+  final Map<String, dynamic> discountMap = <String, dynamic>{
+    'price': 1.0,
+    'numberOfPeriods': 1,
+    'paymentMode': 1,
+    'subscriptionPeriod': subMap,
+  };
+  final Map<String, dynamic> productMap = <String, dynamic>{
+    'productIdentifier': 'id',
+    'localizedTitle': 'title',
+    'localizedDescription': 'description',
+    'currencyCode': 'USD',
+    'downloadContentVersion': 'version',
+    'subscriptionGroupIdentifier': 'com.group',
+    'price': 1.0,
+    'downloadable': true,
+    'downloadContentLengths': <int>[1, 2],
+    'subscriptionPeriod': subMap,
+    'introductoryPrice': discountMap,
+  };
+
+  setUpAll(() {
+    Channel.override = SystemChannels.platform_views;
+    SystemChannels.platform_views.setMockMethodCallHandler(
+        fakePlatformViewsController.fakePlatformViewsMethodHandler);
+  });
+
+  group('canMakePayments', () {
+    test(
+        'SKProductSubscriptionPeriodWrapper should have property values consistent with map',
+        () {
+      final SKProductSubscriptionPeriodWrapper wrapper =
+          SKProductSubscriptionPeriodWrapper.fromMap(subMap);
+      expect(wrapper.numberOfUnits, 0);
```

Comment:
nit: here and throughout it'll be less brittle and a little clearer if you refer to the expected values directly instead of hardcoding magic constants for them.

For example: `expect(wrapper.numberOfUnits, subMap['numberOfUnits']);`

That way if you update the expected value the tests still pass, and the person reading these doesn't have to scroll up to where the expected values are defined to make sure that the tests aren't passing based on a false positive.